tools/linter/adapters/perf_linter.py

```diff
@@ -205,6 +240,48 @@ def lint_empty_constructors(
             return  # only one lint per line
 
 
+def is_constant(tok: TokenInfo) -> bool:
+    return (
+        tok.type == token.NUMBER
+        or (tok.type == token.STRING and tok.string[0] in ('"', "'"))
+        or tok.string in {"True", "False", "None", "...", "-", ","}
+    )
+
+
+def lint_const_set_contains(
```

Comment:
Follow the PLR6201 functionality in ruff for this rule. The issue here is that does not work in the general case because all the constant terms must be hashable too. Detecting if custom classes or constants are hashable requires a full typing system and multi-file analysis so it's far more complicated than this simple script suggests. Doubly so since there is no way to add a rule suppression here to my knowledge.